### PR TITLE
HW5: done

### PR DESCRIPTION
**Pull request by iluwaa**

### CODE REVIEW

File: hw5/pkg/players.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+func NewPlayer(symbol string) *Player {
+	return &Player{
+		Symbol: symbol,
+	}
+}
+
+func (player *Player) ChangeTurn() {
+	if player.Turn == true {
```
Comment (fotonmoton) [inline hw5/pkg/players.go]: You can simplify with: `player.Turn = !player.Turn`

File: hw5/pkg/game.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+			}
+
+		}
+	}
+
+}
+
+func getInput() byte {
+	t, _ := term.Open("/dev/tty")
```
Comment (fotonmoton) [inline hw5/pkg/game.go:134]: As I understand this is for preventing characters to be printed out to the terminal. Why not to use package from go experimental std? https://pkg.go.dev/golang.org/x/term

File: hw5/pkg/game.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+func (game *Game) ConfigureGame() {
+	for index, symbol := range Players() {
+		game.Players = append(game.Players, *NewPlayer(symbol))
+		if index == 0 {
+			game.Players[0].ChangeTurn()
+		}
+	}
+	game.Board = *NewBoard(3)
```
Comment (fotonmoton) [inline hw5/pkg/game.go:43]: I think you want to store pointer to a board. Currently you are copying entire struct on assignment

File: hw5/pkg/game.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+		Title: "TicTacToe, 'X' starts the game.\n" +
+			"Press Enter to make turn.\n" +
+			"Press ESC to exit.",
+		Players: make([]Player, 0),
+	}
+}
+
+func (game *Game) ConfigureGame() {
+	for index, symbol := range Players() {
+		game.Players = append(game.Players, *NewPlayer(symbol))
```
Comment (fotonmoton) [inline hw5/pkg/game.go:38]: Same as with board: you are appending copies of created players. To mitigate this you can store slice of Player pointers

File: hw5/pkg/board.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+			fmt.Printf(" %s%s", printCell, delimeter)
+		}
+	}
+}
+
+func (board *Board) makeTurn(symbol string) {
+
+	for {
+		keyCode := getInput()
+		if keyCode == escape {
```
Comment (fotonmoton) [inline hw5/pkg/board.go:67]: It would be more readable if you will use "switch-case"

**Review by fotonmoton — CHANGES_REQUESTED** [review]

File: hw5/pkg/game.go (hunk 2 above)
Comment (iluwaa) [reply hw5/pkg/game.go:134]: to be honest all this code based on [this](https://medium.com/@nexidian/writing-an-interactive-cli-menu-in-golang-d6438b175fb6) example, I didnt research ant alternatives.

File: hw5/pkg/board.go (hunk 5 above)
Comment (iluwaa) [reply hw5/pkg/board.go:67]: I will keep it in mind next time.

**Pull request merged**